### PR TITLE
SLS Impl: Including asset type filter in logs service.

### PR DESCRIPTION
**Pull request by supritat**
### Requirements

* Changes to /logs API to get logs based on the asset Type Filter 

### Description of the Change

* Including "asset_type" as optional param and valid asset types are mentioned in the config files.

### Benefits

Logs will be fetched for the requested asset_type(api logs or function logs)

### CODE REVIEW

File: core/jazz_logs/config/dev-config.json [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
-        }
-    ]
-}
\ No newline at end of file
+  "BASE_URL": "https://{inst_elastic_search_hostname}/",
+  "DEFAULT_SIZE": 1000,
+  "DEFAULT_TIME_IN_DAYS": 7,
+  "VALID_ENVIRONMENTS": ["prod", "stg", "dev"],
+  "VALID_CATEGORIES": ["api", "function"],
+  "VALID_ASSET_TYPES": ["apilogs", "applicationlogs"],
```
Comment (devsatishm) [inline core/jazz_logs/config/dev-config.json]: I think asset_types should be similar to what we use at other places. These are valid asset types - https://github.com/tmobile/jazz/blob/sls_app/core/jazz_assets/config/global-config.json#L25

And, there has to be a mapping b/w asset_type & log index that we use in ES. 
For ex: 
`asset_type: lambda -> es_indes: applicationlogs/functionlogs`
`asset_type: apigateway -> es_indes: apilogs`

File: core/jazz_logs/index.js [hunk 2]
```diff
@@ -106,14 +111,7 @@ module.exports.handler = (event, context, cb) => {
...
 
-			var servCategory = [];
-
-			if (categoryType.toLowerCase() == 'api') {
-				servCategory = ["apilogs", "applicationlogs"];
-			} else if (categoryType.toLowerCase() == 'function') {
-				servCategory = ["applicationlogs"];
-			}
-
+			var servCategory = assetType ? [`${assetType}`] : [];
```
Comment (devsatishm) [inline core/jazz_logs/index.js]: Configure the mapping b/w asset_type & ES index. Use the index mapped with user provided asset_type.

File: core/jazz_logs/swagger/swagger.json [hunk 3]
```diff
@@ -429,22 +427,30 @@
...
             "api",
             "function"
           ],
           "type": "string",
           "description": "Service type"
         },
-		"start_time": {
+        "asset_type": {
+          "enum": [
+            "apilogs",
```
Comment (devsatishm) [inline core/jazz_logs/swagger/swagger.json]: Supported asset_types: https://github.com/tmobile/jazz/blob/sls_app/core/jazz_assets/config/global-config.json#L25

File: core/jazz_logs/config/dev-config.json (hunk 1 above)
Comment (supritat) [reply core/jazz_logs/config/dev-config.json]: Included list of asset types and asset-index mapping as mentioned.

File: core/jazz_logs/index.js (hunk 2 above)
Comment (supritat) [reply core/jazz_logs/index.js]: Updated

File: core/jazz_logs/swagger/swagger.json (hunk 3 above)
Comment (supritat) [reply core/jazz_logs/swagger/swagger.json]: Updated asset_type list in swagger file

File: core/jazz_logs/config/dev-config.json (hunk 1 above)
Comment (raghits) [reply core/jazz_logs/config/dev-config.json]: changes look good.

File: core/jazz_logs/index.js (hunk 2 above)
Comment (raghits) [reply core/jazz_logs/index.js]: changes look good.

File: core/jazz_logs/swagger/swagger.json (hunk 3 above)
Comment (raghits) [reply core/jazz_logs/swagger/swagger.json]: changes look good.

File: core/jazz_logs/test/test.js [hunk 4]
```diff
@@ -234,6 +235,26 @@ describe('platform_logs', function() {
     assert.isTrue(acceptCount == 2);
   });
 
+  /*
+  * Given event.body.asset_type that is listed as valid in config, handler() should not inform of asset_type exception
```
Comment (pspani) [inline core/jazz_logs/test/test.js]: update comment and test name to conform to what is being tested

Comment (supritat) [reply core/jazz_logs/test/test.js]: Updated

**Pull request merged**